### PR TITLE
Switch all the cross-guides links to use the canonical website url

### PR DESCRIPTION
**Pull request by gsmet**

### CODE REVIEW

File: docs/src/main/asciidoc/context-propagation.adoc [hunk 1]
```diff
@@ -10,7 +10,7 @@ include::./attributes.adoc[]
 Traditional blocking code uses link:https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/ThreadLocal.html[`ThreadLocal`]
  variables to store contextual objects in order to avoid
 passing them as parameters everywhere. Many Quarkus extensions require those contextual objects to operate
-properly: link:getting-started.html[RESTEasy], link:getting-started.html[ArC] and link:transaction.html[Transaction]
+properly: link:rest-json[RESTEasy], link:cdi-reference[ArC] and link:transaction[Transaction]
```
Comment (Sanne) [inline docs/src/main/asciidoc/context-propagation.adoc]: ah, this one was wrong ?!

Comment (gsmet) [reply docs/src/main/asciidoc/context-propagation.adoc]: Yes, they were pointing to the getting started guide. Better point to the guides of each component.

I fixed another one too, it was pointing to an old `async-message-passing` link (we have a redirection - with a canonical meta!) but better avoid it.

File: docs/src/main/asciidoc/hibernate-orm-panache.adoc [hunk 2]
```diff
@@ -51,7 +51,7 @@ To get started:
 * annotate your entities with `@Entity` and make them extend `PanacheEntity`
 * place your entity logic in static methods in your entities
 
-Follow the link:hibernate-orm.html#setting-up-and-configuring-hibernate-orm-without-persistence-xml-recommended[Hibernate set-up guide for all configuration].
+Follow the link:hibernate-orm#setting-up-and-configuring-hibernate-orm-without-persistence-xml-recommended[Hibernate set-up guide for all configuration].
```
Comment (Sanne) [inline docs/src/main/asciidoc/hibernate-orm-panache.adoc]: I think this specific anchor is broken now - it exists currently but I need to check after my own ORM guides PR if this one still exists with same name.

Not your PR's fault, so let's merge it - I'll send a follow up.

**Review by Sanne — APPROVED** [review]

**Comment by Sanne** [comment]
I cross-checked all links with the current website in production. We might miss some if there's other changes in the pipeline, like the ORM anchor. Looking good, I merge!

File: docs/src/main/asciidoc/hibernate-orm-panache.adoc (hunk 2 above)
Comment (Sanne) [reply docs/src/main/asciidoc/hibernate-orm-panache.adoc]: https://github.com/quarkusio/quarkus/pull/5263

**Pull request merged**